webserver/src/repository/pos.rs

```
@@ -325,11 +355,21 @@ impl PosRepositoryTrait for PosRepository {
     async fn find_rewards_by_address(
         &self,
         address: String,
+        epoch: Option<u64>,
     ) -> Result<Vec<PoSRewardDb>, String> {
         let conn = self.app_state.get_db_connection().await;
 
         conn.interact(move |conn| {
+            let epoch = epoch.map(|e| e as i32).unwrap_or_else(|| {
+                pos_rewards::table
+                    .select(diesel::dsl::max(pos_rewards::epoch))
+                    .first::<Option<_>>(conn)
+                    .unwrap_or(Some(0))
+                    .unwrap_or(0)
```

some of these will be legitimate database errors, better to propagate the errors if possible